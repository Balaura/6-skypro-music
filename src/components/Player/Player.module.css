.player {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-start;
}

.controls {
  display: flex;
  flex-direction: row;
  padding: 0 27px 0 31px;
}

.btnPrev,
.btnPlay,
.btnNext,
.btnRepeat,
.btnShuffle {
	padding: 5px;
	display: -webkit-box;
	display: -ms-flexbox;
	display: flex;
	-webkit-box-align: center;
	-ms-flex-align: center;
	align-items: center;
}



.btnPrev,
.btnNext,
.btnShuffle {
  cursor: pointer;
}

.btnPrev:hover,
.btnNext:hover,
.btnShuffle:hover {
  opacity: 0.3;
}


.btnPrev {
  margin-right: 23px;
}

.btnPrevSvg {
  width: 15px;
  height: 14px;
}

.btnPlay {
  margin-right: 23px;
}

.btnPlaySvg {
  width: 22px;
  height: 20px;
  fill: #d9d9d9;
}

.btnNext {
  margin-right: 28px;
  fill: #a53939;
}

.btnNextSvg {
  width: 15px;
  height: 14px;
  fill: inherit;
  stroke: #d9d9d9;
}

.btnRepeat {
  margin-right: 24px;
  cursor: pointer;
}

.btnRepeat.active {
  fill: #ffffff;
}

.btnRepeat .btnRepeatSvg {
  width: 18px;
  height: 12px;
  fill: transparent;
  stroke: #696969;
  /* stroke: #ffffff; */
}

.btnRepeat.active .btnRepeatSvg {
  fill: transparent;
  stroke: #ffffff;
  /* stroke: #696969; */
}

.btnShuffle {
  display: flex;
  align-items: center;
}

.btnShuffleSvg {
  width: 19px;
  height: 12px;
  fill: transparent;
  stroke: #696969;
}

.btnShuffle.active .btnShuffleSvg {
  fill: transparent;
  stroke: #ffffff;
}