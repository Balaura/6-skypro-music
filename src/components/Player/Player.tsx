"use client";

import React from 'react';
import styles from './Player.module.css';
import { useAudioPlayer } from '@/hooks/useAudioPlayer';

interface PlayerProps {
  audioPlayerState: ReturnType<typeof useAudioPlayer>;
}

const Player: React.FC<PlayerProps> = ({ audioPlayerState }) => {
  const { isPlaying, isLooping, togglePlayPause, toggleLooping, playNextTrack, playPreviousTrack, toggleShuffle, isShuffling } = audioPlayerState;

  return (
    <div className={styles.player}>
      <div className={styles.controls}>
        <div className={styles.btnPrev} onClick={playPreviousTrack}>
          <svg className={styles.btnPrevSvg}>
            <use xlinkHref="img/icon/sprite.svg#icon-prev"></use>
          </svg>
        </div>
        <div className={`${styles.btnPlay} ${styles._btn}`} onClick={togglePlayPause}>
          <svg className={styles.btnPlaySvg}>
            <use xlinkHref={`img/icon/sprite.svg#icon-${isPlaying ? 'pause' : 'play'}`}></use>
          </svg>
        </div>
        <div className={styles.btnNext} onClick={playNextTrack}>
          <svg className={styles.btnNextSvg}>
            <use xlinkHref="img/icon/sprite.svg#icon-next"></use>
          </svg>
        </div>
        <div className={`${styles.btnRepeat} ${styles._btnIcon} ${isLooping ? styles.active : ''}`} onClick={toggleLooping}>
          <svg className={styles.btnRepeatSvg}>
            <use xlinkHref="img/icon/sprite.svg#icon-repeat"></use>
          </svg>
        </div>
        <div className={`${styles.btnShuffle} ${styles._btnIcon} ${isShuffling ? styles.active : ''}`} onClick={toggleShuffle}>
          <svg className={styles.btnShuffleSvg}>
            <use xlinkHref="img/icon/sprite.svg#icon-shuffle"></use>
          </svg>
        </div>
      </div>
    </div>
  );
};

export default Player;