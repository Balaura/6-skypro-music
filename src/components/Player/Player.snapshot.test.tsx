import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Player from './Player';

const mockStore = configureStore([]);

jest.mock('@/hooks/useAudioPlayer', () => ({
  useAudioPlayer: () => ({
    togglePlayPause: jest.fn(),
    toggleLooping: jest.fn(),
    playNextTrack: jest.fn(),
    playPreviousTrack: jest.fn(),
    toggleShuffle: jest.fn(),
    isPlaying: false,
    isLooping: false,
    isShuffling: false,
  }),
}));

describe('Player component', () => {
  it('renders correctly', () => {
    const store = mockStore({
      audioPlayer: {
        isPlaying: false,
        isLooping: false,
        isShuffling: false,
      },
    });

    const tree = renderer.create(
      <Provider store={store}>
        <Player />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});