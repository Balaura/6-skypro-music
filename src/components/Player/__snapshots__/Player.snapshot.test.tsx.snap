// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player component renders correctly 1`] = `
<div
  className="player"
>
  <div
    className="controls"
  >
    <div
      className="btnPrev"
      onClick={[MockFunction]}
    >
      <svg
        className="btnPrevSvg"
      >
        <use
          data-testid="prev-button"
          xlinkHref="/img/icon/sprite.svg#icon-prev"
        />
      </svg>
    </div>
    <div
      className="btnPlay _btn"
      data-testid="play-button"
      onClick={[MockFunction]}
    >
      <svg
        className="btnPlaySvg"
      >
        <use
          xlinkHref="/img/icon/sprite.svg#icon-play"
        />
      </svg>
    </div>
    <div
      className="btnNext"
      data-testid="next-button"
      onClick={[MockFunction]}
    >
      <svg
        className="btnNextSvg"
      >
        <use
          xlinkHref="/img/icon/sprite.svg#icon-next"
        />
      </svg>
    </div>
    <div
      className="btnRepeat _btnIcon "
      data-testid="repeat-button"
      onClick={[MockFunction]}
    >
      <svg
        className="btnRepeatSvg"
      >
        <use
          xlinkHref="/img/icon/sprite.svg#icon-repeat"
        />
      </svg>
    </div>
    <div
      className="btnShuffle _btnIcon "
      data-testid="shuffle-button"
      onClick={[MockFunction]}
    >
      <svg
        className="btnShuffleSvg"
      >
        <use
          xlinkHref="/img/icon/sprite.svg#icon-shuffle"
        />
      </svg>
    </div>
  </div>
</div>
`;
